client/collect.py

```diff
@@ -142,7 +143,10 @@ def run_dir(dir_path, parse_output=True, delete_after_success=False,
         else:
             run_file(*args, **kwargs)
     for thread in threads:
-        thread.join()
+        thread_timeout=(60*60) # timeout after one hour.
```

Comment:
Nitpick: put the `thread_timeout` outside the loop, since it doesn't change on each iteration.